ledger/byzcoin/mod.go

```diff
@@ -48,66 +46,58 @@ type Ledger struct {
 	gossiper   gossip.Gossiper
 	bag        *txBag
 	proc       *txProcessor
-	governance governance
-	consumer   consumer.Consumer
+	governance viewchange.Governance
 	encoder    encoding.ProtoMarshaler
+	txFactory  transactions.TransactionFactory
 	closing    chan struct{}
 	initiated  chan error
 }
 
 // NewLedger creates a new Byzcoin ledger.
-func NewLedger(mino mino.Mino, signer crypto.AggregateSigner, consumer consumer.Consumer) *Ledger {
-	cosi := flatcosi.NewFlat(mino, signer)
+func NewLedger(m mino.Mino, signer crypto.AggregateSigner) *Ledger {
+	inventory := mem.NewInventory()
+	taskFactory, gov := newtaskFactory(m, signer, inventory)
+
+	txFactory := basic.NewTransactionFactory(signer, taskFactory)
 	decoder := func(pb proto.Message) (gossip.Rumor, error) {
-		return consumer.GetTransactionFactory().FromProto(pb)
+		return txFactory.FromProto(pb)
 	}
 
-	proc := newTxProcessor(consumer)
-	gov := governance{
-		inventory:     proc.inventory,
-		rosterFactory: newRosterFactory(mino.GetAddressFactory(), signer.GetPublicKeyFactory()),
-	}
-	consensus := cosipbft.NewCoSiPBFT(mino, cosi, gov)
+	consensus := cosipbft.NewCoSiPBFT(m, flatcosi.NewFlat(m, signer), gov)
 
 	return &Ledger{
-		addr:       mino.GetAddress(),
+		addr:       m.GetAddress(),
 		signer:     signer,
-		bc:         skipchain.NewSkipchain(mino, consensus),
-		gossiper:   gossip.NewFlat(mino, decoder),
+		bc:         skipchain.NewSkipchain(m, consensus),
+		gossiper:   gossip.NewFlat(m, decoder),
 		bag:        newTxBag(),
-		proc:       proc,
+		proc:       newTxProcessor(txFactory, inventory),
 		governance: gov,
-		consumer:   consumer,
 		encoder:    encoding.NewProtoEncoder(),
+		txFactory:  txFactory,
 		closing:    make(chan struct{}),
 		initiated:  make(chan error, 1),
 	}
 }
 
-// GetInstance implements ledger.Ledger. It returns the instance with the given
-// key as of the latest block.
-func (ldgr *Ledger) GetInstance(key []byte) (consumer.Instance, error) {
+// GetValue implements ledger.Ledger.
+func (ldgr *Ledger) GetValue(key []byte) (proto.Message, error) {
 	latest, err := ldgr.bc.GetBlock()
 	if err != nil {
-		return nil, xerrors.Errorf("couldn't read latest block: %v", err)
+		return nil, err
```

Comment:
why removing those error wrappings?

Reply:
Oh that's because I removed the GetInstance which does not really makes sense anymore and I usually write the errors alongside the tests but the tests will come in the PR.